instrumentation/net/http/httptrace/otelhttptrace/clienttrace.go

```diff
@@ -199,14 +288,28 @@ func (ct *clientTracer) tlsHandshakeDone(_ tls.ConnectionState, err error) {
 }
 
 func (ct *clientTracer) wroteHeaderField(k string, v []string) {
-	if ct.span("http.headers") == nil {
+	if ct.useSpans && ct.span("http.headers") == nil {
 		ct.start("http.headers", "http.headers")
 	}
-	ct.root.SetAttributes(attribute.String("http."+strings.ToLower(k), sliceToString(v)))
+	if !ct.addHeaders {
+		return
+	}
+	k = strings.ToLower(k)
+	value := sliceToString(v)
+	if _, ok := ct.redactedHeaders[k]; ok {
+		// redact all non-space runes to 'x'
+		value = strings.Map(func(r rune) rune {
```

Comment:
I'd prefer if this wouldn't leak the length/spaces or waste space with long tokens. Just constant `***` sgtm

Reply:
I have even a stronger point of view. I think in such a case, the attribute should NOT be included at all. This is to ensure we do not leak any security-related internals. 

Reasoning: https://github.com/open-telemetry/opentelemetry-specification/pull/1502#discussion_r587860542

 This would be also more in line with other specification recommendations such us:
- https://github.com/open-telemetry/opentelemetry-specification/blame/main/specification/trace/semantic_conventions/http.md#L71
- https://github.com/open-telemetry/opentelemetry-specification/blame/main/specification/trace/semantic_conventions/database.md#L160

Reply:
For now I have updated to use fixed `****` value.  We can instead drop those annotations completely, I dont really have a preference.  If we drop them, I guess I would change the `WithRedactedHeaders` to `WithIgnoredHeaders` to alllow appending to the fixed list.  We can also swap the default to not collect any headers unless a `WithHeaders` option is specified.  For the edge case of trying to debug authentication-related issues, we could add `WithRedactgnoredHeaders` (or some such name) to keep headers in the spans but always hide the values.

Reply:
I prefer the redacted value instead of ignoring one. Seeing which requests were authenticated and which were not is an important common feature in tracing HTTP traffic and can point to issues.